helix-term/src/commands.rs

```diff
@@ -1528,41 +1529,58 @@ fn extend_to_line_bounds(cx: &mut Context) {
     );
 }
 
-fn delete_selection_impl(reg: &mut Register, doc: &mut Document, view_id: ViewId) {
+enum Operation {
+    Delete,
+    Change,
+}
+
+fn delete_selection_impl(cx: &mut Context, op: Operation, with_reg: bool) {
+    let (view, doc) = current!(cx.editor);
+
     let text = doc.text().slice(..);
-    let selection = doc.selection(view_id);
+    let selection = doc.selection(view.id);
 
-    // first yank the selection
-    let values: Vec<String> = selection.fragments(text).map(Cow::into_owned).collect();
-    reg.write(values);
+    if with_reg {
```

Comment:
Let's make a "black hole" register `'_'`. Then the extra parameter can be removed, and this condition can be `if cx.register == Some('_')`. You can then explicitly set the register in the `noyank` versions